garak/probes/generic.py

```diff
@@ -0,0 +1,80 @@
+"""
+**Generic Probe**
+
+This probe can be used to test custom prompts (optionally enhanced with buffs) against target model
+"""
+import logging
+from pathlib import Path
+from garak import _config
+import garak.probes
+from garak.data_sources import DataLoader
+
+
+class CustomPrompts(garak.probes.Probe):
+    """A flexible probe that accepts prompts from external sources.
+    
+    Unlike specialized probes that have hardcoded prompts and detectors,
+    CustomPrompts allows you to:
+    - Load prompts from files (.txt, .json)
+    - Load prompts from HTTP(S) URLs
+    - Use any existing detector
+    """
+    
+    
+    primary_detector = None # Intentionally set to None - users must specify detector(s) explicitly when using this probe
+    extended_detectors = ["always.Fail"] # passthru detector for tests
+    
+    lang = "*"
+    active = False
+    tags = []
+    test_prompts_file = Path(__file__).parent.parent / "resources" / "generic" / "custom_prompts_simple.json"
```

Comment:
As noted this should come from the data directory:

```suggestion
    from garak.data import path as data_path
    test_prompts_file = data_path / "generic" / "custom_prompts_simple.json"
```